pkg/controller/core/core.go

```diff
@@ -52,21 +53,35 @@ func SetupControllers(mgr ctrl.Manager, qManager *queue.Manager, cc *cache.Cache
 		fairSharingEnabled = cfg.FairSharing.Enable
 	}
 
-	cohortRec := NewCohortReconciler(mgr.GetClient(), cc, qManager, CohortReconcilerWithFairSharing(fairSharingEnabled))
-	if err := cohortRec.SetupWithManager(mgr, cfg); err != nil {
-		return "Cohort", err
+	var cqRec *ClusterQueueReconciler
+	if features.Enabled(features.HierarchialCohorts) {
+		cohortRec := NewCohortReconciler(mgr.GetClient(), cc, qManager, CohortReconcilerWithFairSharing(fairSharingEnabled))
+		if err := cohortRec.SetupWithManager(mgr, cfg); err != nil {
+			return "Cohort", err
+		}
+
+		cqRec = NewClusterQueueReconciler(
```

Comment:
Please remove the duplication for calling `NewClusterQueueReconciler`. One idea is to just calculate the list of watchers passed to `WithWatchers` depending on the feature gate.